### PR TITLE
Script to rollback ci-operator

### PR DESCRIPTION
**Pull request by deepsm007**
/cc @openshift/test-platform

### CODE REVIEW

File: hack/rollback_ci-operator.sh [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    if [ $mirror_status -eq 0 ]; then
+        echo "Image mirroring completed successfully!"
+    else
+        echo "Image mirroring failed. Please check the logs for details."
+        exit 1
+    fi
+
+    # Clean up credentials
+    echo "Cleaning up credentials..."
+    rm -f "$creds_file"  # Remove the temporary credentials file
```
Comment (danilo-gemoli) [inline hack/rollback_ci-operator.sh:70]: Isn't this `rm` a duplicate of the in `trap` from above?

File: hack/rollback_ci-operator.sh [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    if [ $? -ne 0 ]; then
+        echo "Error: Failed to tag the digest to ci-operator:latest."
+        exit 1
+    fi
+}
+
+# Function to mirror the ci-operator:latest image to QCI
+mirror_image() {
+    local creds_file=$(mktemp /tmp/registry-push-credentials.XXXXXX.c)
+    trap "rm -f $creds_file" EXIT  # Ensure creds file is removed
```
Comment (danilo-gemoli) [inline hack/rollback_ci-operator.sh:47]: Credentials forgotten somewhere on the filesystem concerns me a bit. Does `trap ...` work even when the scripts raises an unexpected error?

File: hack/rollback_ci-operator.sh [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    # Clean up credentials
+    echo "Cleaning up credentials..."
+    rm -f "$creds_file"  # Remove the temporary credentials file
+}
+
+check_requirements
+
+if [ $# -eq 0 ]; then
+    echo "No digest provided. Rolling back to the previous digest."
+    prev_digest=$(get_tags | yq '.[-2].items[0].image')  # Assuming the second-to-last tag is the "previous" one
```
Comment (danilo-gemoli) [inline hack/rollback_ci-operator.sh:77]: Could you make sure the array length is `>= 3` at least?

File: hack/rollback_ci-operator.sh (hunk 1 above)
Comment (deepsm007) [reply hack/rollback_ci-operator.sh:70]: yeah

**Pull request merged**